### ISSUE TITLE
Custom Search Engines

### ISSUE THREAD

**Issue by JesFEREM** [dmlls/yang#8]
I want to use a search engine self hosted by a friend of mines which is searx.absturztau.be 
it would be cool to add an optional permission to intercept links from any site of your choice 
Here's an example from Extension UBlacklist 
![image](https://github.com/dmlls/yang/assets/60802112/f6e325ed-48b6-43cd-b153-84e60811539a)

**Comment by dmlls** [comment]
Hi @JesFEREM, that's a great idea and definitely should be implemented! It would also help keep the "default" list of permissions smaller and avoid having to update the SearX(NG) instances manually.

Since this will take a bit longer to implement (there are some rather big changes needed, such as restructuring the way settings are stored), for now I've published a [minor release](https://github.com/dmlls/yang/releases/tag/v0.9.21) updating the SearX(NG) instances (including your friend's ;).

I'll leave the issue open until it's implemented, and get back to you then.

**Comment by dmlls** [comment]
Hi back @JesFEREM. After digging a bit into the matter, I've got to know that optional permissions must be declared in the `manifest.json` before they can be requested through the [`permissions` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/permissions/request).

This, unfortunately, makes this feature unable to be implemented in this way, since the manifest cannot be modified programmatically. I've been also taking a look at the example you gave, and the search engine URLs are hardcoded in [`search-engines.ts`](https://github.com/iorate/ublacklist/blob/af221a82657bd6d84305862ea16b4c1e711273ea/src/common/search-engines.ts), which then are injected into [`manifest.json.ts`](https://github.com/iorate/ublacklist/blob/af221a82657bd6d84305862ea16b4c1e711273ea/src/manifest.json.ts).

I can see three ways to work around this:
1. Keep the current model, curating the supported search engine URLs manually, and accepting requests via issues (we could add a shortcut to the addon UI to make it simple for any user).
2. Requesting the `<all_urls>` permission, which I really would not want to do, since it can raise privacy concerns in the users.
3. There is also the `activeTab` permission, which seems like a reasonable compromise between options 1 and 3.

Unfortunately, since we capture the bangs from the URL by looking e.g. at the [query parameters](https://github.com/dmlls/yang/blob/d7b7810f44987c6bb181bb579513bfe6a67f89a5/background.js#L74), options 1 and 3 will lead to accidental triggers since there are plenty of non-search sites using these parameters. Which takes us back to the beginning: the necessity to hardcode the search engine sites (option 1).

Long story short, **I do not see a way of implementing this**, but do please let me know if you have any ideas.
